knotx-core/knotx-template-engine/src/test/java/com/cognifide/knotx/engine/placeholders/UriTransformerReplaceTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Knot.x - Reactive microservice assembler - Templating Engine Verticle
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.engine.placeholders;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.jsoup.nodes.Attribute;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.cognifide.knotx.api.TemplateEngineRequest;
+import com.cognifide.knotx.engine.service.ServiceEntry;
+
+import io.vertx.rxjava.core.MultiMap;
+
+@RunWith(Parameterized.class)
+public class UriTransformerReplaceTest {
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                // path
+                {"/path.html?path={uri.path}", "/a/b/c/d.s1.s2.html/c/d.s.txt#f",
+                        "/path.html?path=/a/b/c/d"},
+                {"/path.html?path={uri.path}", "/a/b/c/d.s1.s2.html/c/d.s.txt#f",
```

Comment:
This is AEM friendly (_all before something with selectors_) but not compatible with [RFC 3986](https://tools.ietf.org/html/rfc3986#section-3)!

This is misleading / incorrect for project not related to AEM.
What do you think?